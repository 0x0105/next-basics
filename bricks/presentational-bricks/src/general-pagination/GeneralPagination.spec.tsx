import React from "react";
import { shallow, mount } from "enzyme";
import { GeneralPagination } from "./GeneralPagination";

describe("GeneralPagination", () => {
  it("should work", () => {
    const handleOnChange = jest.fn();
    const onShowSizeChange = jest.fn();
    const wrapper = shallow(
      <GeneralPagination
        page={1}
        pageSize={20}
        total={50}
        handleOnChange={handleOnChange}
        onShowSizeChange={onShowSizeChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should handleOnChange to have been called ", () => {
    const handleOnChange = jest.fn();
    const onShowSizeChange = jest.fn();
    const wrapper = mount(
      <GeneralPagination
        page={1}
        pageSize={20}
        total={50}
        handleOnChange={handleOnChange}
        onShowSizeChange={onShowSizeChange}
      />
    );
    wrapper
      .find("Pager")
      .at(1)
      .simulate("click");
    expect(handleOnChange).toHaveBeenCalled();
  });
});
