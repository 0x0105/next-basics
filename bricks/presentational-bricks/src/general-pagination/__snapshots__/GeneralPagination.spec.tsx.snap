// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralPagination should work 1`] = `
<Pagination
  className=""
  current={1}
  onChange={[MockFunction]}
  onShowSizeChange={[MockFunction]}
  pageSize={20}
  pageSizeOptions={
    Array [
      "10",
      "20",
      "50",
    ]
  }
  showSizeChanger={true}
  showTotal={[Function]}
  total={50}
/>
`;
